src/backend/distributed/operations/shard_split.c

```diff
@@ -2238,7 +2291,8 @@ GetNextShardIdForSplitChild()
 	}
 
 	shardId = SafeStringToUint64(PQgetvalue(result, 0, 0 /* nodeId column*/));
-	CloseConnection(connection);
+	PQclear(result);
```

Comment:
now that we no longer close the connection, we need to be careful that we don't cache it for the lifetime of the session.

I suggest introducing a GetLocalConnectionForSubtransactions that opens a connection to localhost with OUTSIDE_TRANSACTION and sessionLifespan set to false.

Reply:
Ack. Will do in another PR.

Reply:
@marcocitus , I don't see any session flag in 'GetNodeUserDatabaseConnection'. The only thing I could find was 'start_session_level_connection_to_node' matching 'sessionLifespan'. What exactly are you referring to here?

Reply:
Discussed offline. This has been renamed to 'forceCloseAtTransactionEnd'.